### ISSUE TITLE
Is it necessary to undo the changes to sysctl when you are done?

### ISSUE THREAD

**Issue by snoblenet** [mitmproxy/mitmproxy#1105]
Steps like `sudo sysctl -w net.inet.ip.forwarding=1` seem significant.

Is it necessary for security or other reasons to undo any of these changes once you're done with mitmproxy?

**Comment by mhils** [comment]
Thanks for asking!

TL;DR: I don't think so.

`net.inet.ip.forwarding` enables traffic to travel between two interfaces on said machine. For example, if you set `net.inet.ip.forwarding=0` on your router/gateway, you won't have internet anymore, but can still communicate with your local peers.

If you are running mitmproxy on a normal machine, which typically is just connected to the local WiFi or LAN, this change should (as I understand it) not affect your system's security in any way. It just instructs your machine to take packets addressed to your MAC address but to a different IP address and reinsert them to the network (this time with either the target IP's MAC address or, if it goes to the internet, the Router/Gateway's MAC address). http://docs.mitmproxy.org/en/latest/modes.html#transparent-proxy has some illustrations.

If you run this on a machine which for some reason has interfaces to two otherwise isolated networks, we assume you understand the implications of this command. Last but not least, `sysctl -w` does not survive a reboot.
